mlir/lib/Conversion/GPUToLLVMSPV/GPUToLLVMSPV.cpp

```diff
@@ -259,23 +258,33 @@ struct GPUShuffleConversion final : ConvertOpToLLVMPattern<gpu::ShuffleOp> {
   }
 
   /// Get the subgroup size from the target or return a default.
-  static int getSubgroupSize(Operation *op) {
-    return spirv::lookupTargetEnvOrDefault(op)
-        .getResourceLimits()
-        .getSubgroupSize();
+  static std::optional<int> getSubgroupSize(Operation *op) {
+    auto module = op->getParentOfType<gpu::GPUModuleOp>();
+    if (!module)
+      return std::nullopt;
+    auto spatialExtents = module.getSpatialExtents();
+    if (!spatialExtents)
+      return std::nullopt;
+    return spatialExtents->getReqdSubgroupSize();
```

Comment:
Can we provide a `gpu::lookupSpatialExtents`? This should work as `spirv::lookupTargetEnvOrDefault`, not requiring a `gpu.module` parent. That requirement may not play well on Triton, where we may wanna use the attribute you introduce here as an anonymous attribute on a builtin `module`.

Reply:
right, I guess I kind of forgot that you could just apply the attribute to any operation if you wanted to. I will do that.